src/mono/mono/metadata/class-private-definition.h

```diff
@@ -127,6 +128,9 @@ struct _MonoClass {
 
 	/* Infrequently used items. See class-accessors.c: InfrequentDataKind for what goes into here. */
 	MonoPropertyBag infrequent_data;
+
+	MonoClass **variant_search_table;
+	int variant_search_table_length;
```

Comment:
Can we stick this into the `infrequent_data` property bag?

Reply:
That would make every lookup slower

Reply:
We can leave a bit in `MonoClass` that's set if the table has non-zero length.

Reply:
What I mean is, every time we fetch the table itself we'd have to scan the whole property bag linked list to find it

Reply:
but that should be rare right? we don't do this on every variant interface call?

Reply:
We have to do it on every call based on the spec. minijit caches it with the IMT but interp doesn't have a cache.